db/upgrade.php

```diff
@@ -235,6 +227,35 @@ function xmldb_block_coursefeedback_upgrade($oldversion = 0) {
 
         upgrade_block_savepoint(true, 2024010900, 'coursefeedback');
     }
+
+    if ($oldversion == 2024015000 ) {
+        // Following changes need to be checked only for version '2024015000'.
+
+        $dbman = $DB->get_manager();
+        $table = new xmldb_table('block_coursefeedback_answers');
+
+        // Drop 'textanswer' field in 'block_coursefeedback_answers'.
+        $field = new xmldb_field('textanswer', XMLDB_TYPE_TEXT, 'medium', null, null, null, null, 'answer');
+        $dbman->drop_field($table, $field);
+
+
+        // Check if the 'answer' field in the table 'block_coursefeedback_answers' has the correct 'NOTNULL' value.
+        $field = new xmldb_field('answer', XMLDB_TYPE_INTEGER, '4', null, XMLDB_NOTNULL, null, null, 'questionid');
+        // Drop index temporarily to make field change possible.
+        $index = new xmldb_index('mdl2_bloccouransw_coucouque_ix', XMLDB_INDEX_NOTUNIQUE, ['course', 'coursefeedbackid', 'questionid', 'answer']);
```

Comment:
Der DB Präfix ist doch config abhängig. Ich glaube, du musst den Index Namen hier ohne `mdl2_` übergeben.

Reply:
Danke, erledigt in: [9e674f2](https://github.com/eberhardt/moodle-block_coursefeedback/pull/44/commits/9e674f2ffc180e7e4fcabf6149fe8781800346fb)